analyses/survival-analysis/survival-analysis_immune.Rmd

```diff
@@ -0,0 +1,324 @@
+
+## Analysis: quantiseq cell type fractions and PDL-1  across entire cohort
+
+> No effects are significant
+
+```{r cox_cell_type_cd274}
+fit_save_model(combined_df_stranded,
+               paste0(quantiseq_terms, "+CD274"),
+               file.path(results_dir, "cox_terms_quantiseq.RDS"),
+               model_type = "multivariate")
+```
+
+
+## Analysis: quantiseq cell type fractions and PDL-1 for HGG 
+
+> No effects are significant
+
+```{r cox_hgg_cell_type_cd274}
+hgg_groups <- c("High-grade glioma astrocytoma", "Diffuse midline glioma", "Diffuse intrinsic pontine glioma")
+hgg_df <- combined_df_stranded %>%
+  filter(cancer_group %in% hgg_groups)
+
+fit_save_model(combined_df_stranded,
+               paste0(quantiseq_terms, "+CD274"),
+               file.path(results_dir, "cox_hgg_terms_quantiseq.RDS"),
+               model_type = "multivariate")
+```
+
+
+## Determine suitable cancer groups for analysis
+
+For cancer group analyses, we only want to consider groups with at least THREE DECEASED, again considering stranded only. We find those groups here:
+```{r cancer_groups}
+suitable_cancer_groups <- combined_df_stranded %>%
+  drop_na() %>%
+  count(cancer_group,OS_status) %>%
+  spread(OS_status, n,
+         # When there are no events, record the event count as 0 instead of the default NA
+         fill = 0) %>%
+  filter(DECEASED >= 3) 
+
+# show result: there are 5 groups to keep
+suitable_cancer_groups
+```
+
+For molecular signatures analysis, we want to only consider cancer groups, of those identified above, that have more than 1 molecular subtype.
+
+```{r suitable_subtypes}
+combined_df_stranded %>%
+  filter(cancer_group %in% suitable_cancer_groups$cancer_group) %>%
+  select(cancer_group, molecular_subtype) %>%
+  distinct() %>%
+  # count number of subtypes
+  count(cancer_group) %>%
+  filter(n > 1) -> suitable_subtypes
+
+
+# n is how many subtypes the group has. There needs to be more than 1.
+suitable_subtypes
+```
+
+
+
+## Analysis: quantiseq cell type fractions and PDL-1 for each cancer group of interest
+
+> Most models here are not robust. **Only Medulloblastoma** returns no warning messages.
+
+```{r cell_type_cg}
+
+base_filename <- "cox_per_cg_terms_quantiseq.RDS" # all file names here will follow this convention
+
+for (cg in suitable_cancer_groups$cancer_group) {
+  cat("==============", cg, "==================")
+  # filter to cancer group of interest
+  fit_data <- combined_df_stranded %>% 
+    filter(cancer_group == cg)
+  
+  # Define filename where model will be saved
+  filename <- file.path(
+    results_dir,
+    str_replace(base_filename, "cg", cg)
+  )
+  
+  # Fit model
+  fit_save_model(fit_data, 
+                 paste0(quantiseq_terms, "+CD274"), 
+                 filename, 
+                 model_type = "multivariate") %>% print()
+}
+```
+
+
+
+## Analysis: Differences among molecular subtypes within cancer groups
```

Comment:
This is where I wanted to see the HGG group...

here, we can either show just for notebook purposes, the suitable _non-HGG_ cancer groups + the HGG group, or just run the HGG group (sans oligo) doing a univariate analysis for molecular subtype.

Reply:
I'm still not understanding exactly what you mean with the HGG group analysis. I did look for molecular subtype differences within HGG groups, and I don't understand how this request is different from the analysis I performed, and based on [this comment] I understood that testing non-HGG groups was not of interest. 

It would help a lot if you could tell me the exact formula you want to see, or alternatively you are welcome to add commits directly to this branch.

Reply:
Will do in a new PR!